pkg/controllers/providers/instancetype/memoryoverhead/controller.go

```diff
@@ -0,0 +1,77 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package memoryoverhead
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/karpenter-provider-aws/pkg/providers/instancetype"
+	"github.com/awslabs/operatorpkg/reasonable"
+	corev1 "k8s.io/api/core/v1"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	"sigs.k8s.io/karpenter/pkg/operator/injection"
+	"time"
+)
+
+type Controller struct {
+	kubeClient           client.Client
+	instancetypeProvider *instancetype.DefaultProvider
+}
+
+func NewController(kubeClient client.Client, instancetypeProvider *instancetype.DefaultProvider) *Controller {
+	return &Controller{
+		kubeClient:           kubeClient,
+		instancetypeProvider: instancetypeProvider,
+	}
+}
+
+func (c *Controller) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	ctx = injection.WithControllerName(ctx, "instancetypes.memoryoverhead")
+	if err := c.instancetypeProvider.UpdateInstanceTypeMemoryOverhead(ctx, c.kubeClient); err != nil {
+		return reconcile.Result{}, fmt.Errorf("updating instancetype, %w", err)
+	}
+	return reconcile.Result{RequeueAfter: 12 * time.Hour}, nil
```

Comment:
That's mostly to trigger refreshing of the cache before expiration. The function here doesn't actually use the specific node info since UpdateInstanceTypeMemoryOverhead() still lists out nodes on it's own.

Reply:
The request we're requeuing is for a specific Node though. This results in an individual reconciliation per registered node every 12 hours.

Reply:
Should be fixed now